aten/src/ATen/test/vec256_test_all_types.cpp

```diff
@@ -0,0 +1,1098 @@
+        ValueType<vec_type> case1 = -658.5f;
+        ValueType<vec_type> exp1 = -658.f;
+        ValueType<vec_type> case2 = -657.5f;
+        ValueType<vec_type> exp2 = -658.f;
+        auto test_case =
+            TestingCase<vec_type>::getBuilder()
+            .addDomain(CheckWithinDomains<UVT>{ { {-1000, 1000}} })
+            .addCustom({ {case1},exp1 })
+            .addCustom({ {case2},exp2 })
+            .setTrialCount(64000);
+        test_unary<vec_type>(
+            "round", RESOLVE_OVERLOAD(at::native::round_impl),
+            [](vec_type v) { return v.round(); }, test_case);
+    }
+
+    TYPED_TEST(Rounding, Ceil) {
+        using vec_type = TypeParam;
+        using UVT = UvalueType<TypeParam>;
+        test_unary<vec_type>(
+            "ceil", RESOLVE_OVERLOAD(std::ceil),
+            [](vec_type v) { return v.ceil(); });
+    }
+
+    TYPED_TEST(Rounding, Floor) {
+        using vec_type = TypeParam;
+        test_unary<vec_type>(
+            "floor", RESOLVE_OVERLOAD(std::floor),
+            [](vec_type v) { return v.floor(); });
+    }
+
+    TYPED_TEST(Rounding, Trunc) {
+        using vec_type = TypeParam;
+        test_unary<vec_type>(
+            "trunc", RESOLVE_OVERLOAD(std::trunc),
+            [](vec_type v) { return v.trunc(); });
+    }
+
+    TYPED_TEST(SqrtAndReciprocal, Sqrt) {
+        //pytorch complex sqrt precision differs from std 
+        //we will check for error under 0.001 
+        using vec_type = TypeParam;
+        using UVT = UvalueType<TypeParam>;
+        auto test_case =
+            TestingCase<vec_type>::getBuilder()
+            .addDomain(CheckWithinDomains<UVT>{ { {-100, 100}}, true, 1.e-3f})
+            .setTrialCount(200);
+        test_unary<vec_type>(
+            "sqrt", RESOLVE_OVERLOAD(std::sqrt),
+            [](vec_type v) { return v.sqrt(); }, test_case);
+    }
+
+    TYPED_TEST(SqrtAndReciprocalReal, Sqrt) {
+        using vec_type = TypeParam;
+        test_unary<vec_type>(
+            "sqrt", RESOLVE_OVERLOAD(std::sqrt),
+            [](vec_type v) { return v.sqrt(); }, false, {}, true);
+    }
+
+    TYPED_TEST(SqrtAndReciprocalReal, RSqrt) {
+        using vec_type = TypeParam;
+        test_unary<vec_type>(
+            "rsqrt", rsqrt<ValueType<vec_type>>,
+            [](vec_type v) { return v.rsqrt(); }, false,
+            RESOLVE_OVERLOAD(filter_zero));
+    }
+
+    TYPED_TEST(SqrtAndReciprocalReal, Reciprocal) {
+        using vec_type = TypeParam;
+        test_unary<vec_type>(
+            "reciprocal",
+            reciprocal<ValueType<vec_type>>,
+            [](vec_type v) { return v.reciprocal(); }, false,
+            RESOLVE_OVERLOAD(filter_zero));
+    }
+
+    TYPED_TEST(Trigonometric, Sin) {
+        using vec_type = TypeParam;
+        using UVT = UvalueType<TypeParam>;
+        auto test_case =
+            TestingCase<vec_type>::getBuilder()
+            .addDomain(CheckWithinDomains<UVT>{ { {-4096, 4096}}, true, 1.2e-7f})
```

Comment:
I implemented sin from avx_mathfn and later modified avx_mathfun a little and then tested against cephes peaks. 
[cephes doc](https://www.netlib.org/cephes/singldoc.html) 
But I suggest to move trigonometric and other such functions to the separate headers and implement it porting sleef code. it will be easier and work the same for all architectures

Reply:
here is my suggestion as I did not understand the wisdom behind making such functions as a member operations
https://github.com/pytorch/pytorch/pull/41541#issuecomment-671120195